### PR TITLE
[RSDK-8144]   Add CI workflow tests for ultrasonic mod

### PR DESCRIPTION
**Pull request by hexbabe**
[RSDK-8144](https://viam.atlassian.net/browse/RSDK-8144)

### CODE REVIEW

File: ultrasonic/ultrasonic_camera_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+	logger := logging.NewTestLogger(t)
+
+	// Test newCamera
+	cam, err := newCamera(context.Background(), deps, conf, logger)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, cam, test.ShouldNotBeNil)
+
+	// Test Close
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```
Comment (randhid) [inline ultrasonic/ultrasonic_camera_test.go:50]: Do we have go routines started in the ultrasonic sensor? Let's check they're closed correctly. https://engineering.razorpay.com/detecting-goroutine-leaks-with-test-cases-b0f8f8a88648
Olivia can help set up this test, she's done it for failover components.

**Review by randhid — APPROVED** [review]
one nit! looks good!

File: .github/workflows/quality-checks.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    
+    - name: Set up Go
+      uses: actions/setup-go@v3
+      with:
+        go-version: 1.22
+
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y libnlopt-dev
```
Comment (seanavery) [inline .github/workflows/quality-checks.yml:30]: ugh, our `go.viam.com/test` dep uses nlopt

File: ultrasonic/ultrasonic_sensor_test.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	// Test newSensor
+	s, err := newSensor(context.Background(), deps, conf, logger)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, s, test.ShouldNotBeNil)
+
+	us, ok := s.(*usSensor)
+	test.That(t, ok, test.ShouldBeTrue)
+
+	// Test Close
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```
Comment (seanavery) [inline ultrasonic/ultrasonic_sensor_test.go:100]: Assuming 2 seconds is safe here to allow for Close to finish without introducing flakiness.

**Review by seanavery — APPROVED** [review]
LGTM

File: .github/workflows/quality-checks.yml (hunk 2 above)
Comment (randhid) [reply .github/workflows/quality-checks.yml:30]: .... do we need to use our own test package? This seems ridiculous.

Comment (hexbabe) [reply .github/workflows/quality-checks.yml:30]: Oh that's why I needed that haha...

File: ultrasonic/ultrasonic_sensor_test.go (hunk 3 above)
Comment (hexbabe) [reply ultrasonic/ultrasonic_sensor_test.go:100]: Should be since these are just unit tests

File: ultrasonic/ultrasonic_camera_test.go (hunk 1 above)
Comment (hexbabe) [reply ultrasonic/ultrasonic_camera_test.go:50]: I'm fairly sure `s.board.StreamTicks` does

File: .github/workflows/quality-checks.yml (hunk 2 above)
Comment (randhid) [reply .github/workflows/quality-checks.yml:30]: but also how is it gettign nlopt? https://github.com/viamrobotics/test, this seems like it should not need it at all

File: ultrasonic/ultrasonic_camera_test.go (hunk 1 above)
Comment (hexbabe) [reply ultrasonic/ultrasonic_camera_test.go:50]: Added! Thanks @oliviamiller

**Review by randhid — APPROVED** [review]
Can we quickly figure out why it needs nlopt? We're trying to remove th requirement of that package when you import arms in go, but I'm weirded out by the test package needing it, it seems to be a bunch of assertions/

**Comment by randhid** [comment]
> Can we quickly figure out why it needs nlopt? We're trying to remove th requirement of that package when you import arms in go, but I'm weirded out by the test package needing it, it seems to be a bunch of assertions/

Not blocking merging but maybe ask in engineering or ask abe what's pulling it in?

**Comment by hexbabe** [comment]
> Can we quickly figure out why it needs nlopt? We're trying to remove th requirement of that package when you import arms in go, but I'm weirded out by the test package needing it, it seems to be a bunch of assertions/

Current hypothesis is that its from inject board? Or some other test injections we make. Asked in team eng tho. Gonna merge. Thanks for review and helpful question

**Pull request merged**